src/NHibernate/Linq/Functions/StringGenerator.cs

```diff
@@ -275,6 +276,20 @@ public override HqlTreeNode BuildHql(MethodInfo method, Expression targetObject,
 				return treeBuilder.MethodCall("trim", treeBuilder.Ident(trimWhere), treeBuilder.Constant(trimChars), treeBuilder.Ident("from"), visitor.Visit(targetObject).AsExpression());
 			}
 		}
+
+		private static string ExtractTrimChars(IReadOnlyList<Expression> arguments)
+		{
+			if (arguments.Count > 0)
+			{
+				var constantExpression = (ConstantExpression) arguments[0];
+				if (constantExpression.Value is char c)
+					return c.ToString();
```

Comment:
This code path is an addition compared to previous code. Is this an unrelated bug fix?